langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/statements/foreach/ForeachNegativeTests.java

```diff
@@ -104,6 +104,8 @@ public void testForeachSemanticsNegative() {
                 "mismatch with member type count", 342, 13);
         BAssertUtil.validateError(compile, index++, "invalid list binding pattern; member variable count " +
                 "mismatch with member type count", 346, 18);
+        BAssertUtil.validateError(compile, index++, "invalid error variable; expecting an error type " +
```

Comment:
Did we create an issue to track allowing compatible binding patterns with unions?

Reply:
ya https://github.com/ballerina-platform/ballerina-lang/issues/40377